### PR TITLE
CookieCutter conversion

### LINKED ISSUE #2: Should we use CookieCutter instead?
I'm assuming this is a cookicutter project. If this is the case I don't think the current set up works. You will need to nest the project into another folder. The top level is meant to cover all the cookie cutter meta data and the folder with `{{project.name}}` contains all the stuff that will be generated when calling cookie cutter.

### PR DESCRIPTION
**Pull request by AntoineDao**
Here's a proposed change to the template repository setup so that it works with cookiecutter. The README.md at the root of the repo should be self-explanatory enough to enable pretty much anyone to generate their own ladybug-tools repo from scratch.

@mostaphaRoudsari I can't add you as a reviewer here so I'll just nudge you with an `@` instead :wink: 

resolves #2

### CODE REVIEW

**Comment by AntoineDao** [comment]
Here is a quickstart to test this PR before merging it:

## QuickStart
1. Install [cookiecutter](https://cookiecutter.readthedocs.io/en/latest/readme.html) and [pygithub](https://pygithub.readthedocs.io/en/latest/introduction.html) (to automate creating the repositories on Github after generating them locally)
```console
pip install cookiecutter pygithub
```
2. Run cookiecutter against this repository and follow the prompts
```console
cookiecutter gh:antoinedao/ladybug-tools-template

# Follow cookiecutter prompts
```

3. If you did not initialise the project on Github do so manually. Please visit https://github.com/new and create a repository for ladybug-tools/YOUR-PROJECT-SLUG (example: ladybug-tools/fire-ant for a project called Fire Ant). If you don't have the correct authorization to create repositories on the ladybug-tools organization then create a repo for your Github user and get in touch with the admins.

4. Activate the repository on [TravisCI](https://travis-ci.org/).
  
  image
  
  and set-up GH_TOKEN and PYPI_PASSWORD.

  image

5. Activate this repository on [coveralls](https://coveralls.io/).

The repository should be ready! Now go write some actual code!

File: cookiecutter.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+  "name": "My New Project",
+  "project_slug": "{{ cookiecutter.name|lower|replace(' ', '-') }}",
+  "pypi_name": "lbt-{{ cookiecutter.project_slug}}",
```
Comment (mostaphaRoudsari) [inline cookiecutter.json]: lbt should only be used for the libraries that packages different libraries together. I think we should remove it from the default name.

File: hooks/post_gen_project.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
+import subprocess
```
Comment (mostaphaRoudsari) [inline hooks/post_gen_project.py:1]: This is cool and is basically what was behind the idea of using a repo as a template.  I think now that we are going to use cookicutter we should disable the template option.

**Review by mostaphaRoudsari — APPROVED** [review]
Looks good! Thank you @AntoineDao. Let's make sure to remove `lbt-` from default package name before you merge this in.

It doesn't take advantage of GitHub template but it does everything that the template is supposed to do and does it even better.

I'm not sure how I feel about the auto deployment set-up right now. We probably should add github account / organization as one of the optional inputs instead of have it hard-coded to ladybug-tools [here](https://github.com/ladybug-tools/ladybug-tools-template/pull/3/files?file-filters%5B%5D=.cfg&file-filters%5B%5D=.md&file-filters%5B%5D=.py&file-filters%5B%5D=.rst&file-filters%5B%5D=.sh&file-filters%5B%5D=.txt&file-filters%5B%5D=.yml&file-filters%5B%5D=No+extension&file-filters%5B%5D=dotfile#diff-6936a93b31eaf599906bf674831050d3R54). In many cases even I do start the library on my personal account and then transfer it to Ladybug Tools. I can imagine other developers to use this repository to initiate a new library on their own repo and start the development and send a transfer request once it is ready to be added to ladybug-tools.

File: hooks/post_gen_project.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+            """
+        )
+
+        return False
+
+
+git_initialised = initialise_git()
+
+if PUSH_TO_GITHUB == 'yes':
+    pushed_to_github = push_to_github(git_initialised)
```
Comment (mostaphaRoudsari) [inline hooks/post_gen_project.py:83]: There is a bug here. push_to_github takes two inputs, you are passing only one.

```
  File "C:\Users\Mostapha\AppData\Local\Temp\tmprexzyd19.py", line 85, in <module>
    pushed_to_github = push_to_github(git_initialised)
TypeError: push_to_github() missing 1 required positional argument: 'git_initialised'
```

**Review by mostaphaRoudsari — CHANGES_REQUESTED** [review]
There is a bug in `post_gen_project.py` which should be easy to fix.

Also let's make sure to remove `lbt-` from the default name.

**Comment by mostaphaRoudsari** [comment]
@AntoineDao, I tried it for `honeybee-radiance-command` repository and the new `.travis.yml` file was giving me errors for no good reason.

https://travis-ci.org/ladybug-tools/honeybee-radiance-command/builds/569051106

Changing it to the old one just worked fine

https://travis-ci.org/ladybug-tools/honeybee-radiance-command/builds/569057636

Otherwise it worked great and was such a joy to use.

File: {{cookiecutter.project_slug}}/README.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+pip install {{cookiecutter.pypi_name}}
+```
+
+## QuickStart
+```python
+import {{ cookiecutter.pkg_name }}
+
+```
+
+## [API Documentation](http://ladybug-tools.github.io/apidoc/{{cookiecutter.project_slug}})
```
Comment (mostaphaRoudsari) [inline {{cookiecutter.project_slug}}/README.md]: This is the wrong link.

```suggestion
## [API Documentation](http://ladybug-tools.github.io/{{cookiecutter.project_slug}}/docs)
```

**Review by mostaphaRoudsari — APPROVED** [review]
Looks great! Thanks. 🛫

**Pull request merged**